### DISCUSSION TITLE
Music Provider Feature Request Poll #3

### DISCUSSION THREAD

**Discussion by OzGav** [music-assistant/support#2356] [Feature Polls]
It has been fourteen months since the last poll and a lot has changed. MA V2.0 is out of beta. Deezer, Jellyfin, Subsonic and Apple Music have been added as Music Providers. So it is time to re-do the poll to find out what the Community would like next!

If you would like to discuss these options in more detail then here are the threads. These are in order of the previous vote numbers so lets see if they stay that way! Do upvote these threads as well as voting here!

Amazon Music https://github.com/orgs/music-assistant/discussions/626
Local Audio In https://github.com/orgs/music-assistant/discussions/1156
Audiobookshelf
Audible https://github.com/orgs/music-assistant/discussions/862
Emby https://github.com/orgs/music-assistant/discussions/583
Pandora https://github.com/orgs/music-assistant/discussions/393
WebDAV https://github.com/orgs/music-assistant/discussions/526
Bandcamp https://github.com/orgs/music-assistant/discussions/2315

The poll only takes eight options but here are some other contenders. Comment and raise your votes there.
SiriusXM https://github.com/orgs/music-assistant/discussions/1161
Mixcloud https://github.com/orgs/music-assistant/discussions/2229

If you want a provider not listed above then open a new feature request and it will be linked above once it starts to get some traction!

**Comment by DutchJaFO** [comment]
podcasts and audiobooks ... would be really nice to have ;)

**Reply by OzGav** [reply]
Yes this is an enabling feature that is coming so the above is for what providers you would like once that is in place.

**Comment by Moldiz** [comment]
I would love to see Bluetooth as well as audio in.

**Reply by OzGav** [reply]
You could use a picoreplayer now as that supports bluetooth

**Reply by Moldiz** [reply]
Yeah, the problem is, I don't use/have raspberry. I'm hosting my Home Assistant and Music Assistant on my multi-purpose linux homelab server. And I spend tons of hours just to modify pipewire to run as system service instead of per user one. Everything is hosted on debian. And streaming with BT outside of MA is at least problematic. Would be cool if MA had such option :)

**Reply by wambs** [reply]
I have a Raspberry Pi Zero 2 W Complete starter kit which cost me $44
I installed piControllerPlayer, works great!!
So, I purchase 2 more to install in order parts of the house.
Bluetooth is limited in range. I have bluetooth devices all over the house.
So, I needed multiple piControllerPlayer's ,in th house to cover all the devices.
This is a much better solution than running it on the local house assistant server.

**Comment by madbrain76** [comment]
My vote is for an option not yet on the list, a DLNA/uPNP music provider.

Several media servers expose a DLNA/uPNP interface. These include :
1) Plex (already supported natively by MA), which I use, but mainly for video, not music,
2) Emby (on the poll list), which I tried, but ruled out - I forget why,
3) Serviio (have not used it)
4) Universal Media server (have not tried it either)
5) Twonky server (have also not used it)
6) JRiver Media Center - what I have been primarily using for 12 years to manage my music library
7) Kodi
8) Windows media streaming (also have not tried)
9) minimserver (have not used it)
10) Logitech Media server (now Lyrion Music server). (also have not tried it)

It seems to me that it would be valuable to have DLNA at least as a poll option, especially since it would open up multiple possibilities for managing one's local music library. I am a bit surprised it hasn't been suggested before.

**Reply by OzGav** [reply]
Discussion exists here https://github.com/orgs/music-assistant/discussions/388

**Reply by madbrain76** [reply]
> Discussion exists here https://github.com/orgs/music-assistant/discussions/388

That discussion is about exposing MA as a DLNA server, which may also be good, but is not the same request as adding a DLNA music provider, which would let MA you fetch and play music content stored in additional server software that I listed.

**Reply by OzGav** [reply]
Then its this discussion then https://github.com/orgs/music-assistant/discussions/890

**Reply by madbrain76** [reply]
> Then its this discussion then https://github.com/orgs/music-assistant/discussions/890

Yes, that is it. Given that the DLNA music provider would also include support for Emby, and would essentially be a superset of that provider, I think it would make sense to add DLNA to the next poll.

**Comment by Hedda** [comment]
Would a AUX input port from an ADC (Analog to Digital Converter) Audio Capture Device also be counted as a source for "Local Audio-In"?

* https://github.com/orgs/music-assistant/discussions/1156

Personally I would really love it if was able to get audio input from example USB audio capture from analog audio sources such as a turntable (LP vinyl record player) into Music Assistant by being able to simply "Local Audio-In" as a music provider. As per my related feature request here:

* https://github.com/orgs/music-assistant/discussions/2343

That would made it much easier and way more user-friendly function than the hoops you have to jump through to get a solution for it today. Ex:

* https://github.com/quebulm/Raspberry-Pi-Vinyl-Streamer

* https://github.com/gieljnssns/darkice-libaacplus-rpi-guide/

**Comment by grainsoflight** [comment]
Ive been looking for a way to cast my USB turntable for years, I really hope local sources becomes a thing. None of the methods I've tried in the past have been reliable, or they did not support lossless formats

**Reply by Hedda** [reply]
I agree. While it is possible to get audio from an analog source using a USB audio capture device and stream it via other workarounds such as an Icecast streaming server there is loads of hops you need jump through and a hassle to maintain

Wish that there was a plug-and-play audio provider for getting audio stream directly from a local audio capture device into Music Assistant which resample, transcode and mux it to best possible supported format.

By the way, see these indirectly related patches which will add a media proxy to Home Assistant core and extend the media player info in ESPHome with a list of supported audio formats which can then be used in Home Assistant to swap out media URLs going to the ESPHome device with proxy URLs that contain the converted audio. That should allow you to at least capture the audio at best possible sample rate and then have Home Assistant automatically convert it to the higest format that the player is capable of:

* https://github.com/home-assistant/core/pull/123254
* https://github.com/esphome/aioesphomeapi/pull/925
* https://github.com/esphome/esphome/pull/7318

Other than that suggest that please check out and join these two dedicated feature request discussions:

* https://github.com/orgs/music-assistant/discussions/1156

and 

* https://github.com/orgs/music-assistant/discussions/2343

Perhaps more active discussions in those will draw the attention and interest of a devleloper with the skills and time to make those ideas happen.

**Comment by vyalekseev** [comment]
Add please Yandex Music Provider https://music.yandex.ru

**Reply by OzGav** [reply]
See related discussion here https://github.com/orgs/music-assistant/discussions/2742

**Comment by mac-city** [comment]
I would like to use MPD as a music provider.

**Reply by 2q2code** [reply]
Same. I currently have it set up to output to a loopback interface and am piping that through snapcast. I organise with beets, and have invested in mpd/mpc apps of good quality (native) on iOS, so I'd love to be able to use them with this system too.

**Reply by Hedda** [reply]
FYI, posted a didicated request discussion thread here:

* https://github.com/orgs/music-assistant/discussions/4230

**Comment by mcleanrs** [comment]
LiveOne
https://www.liveone.com

It's internet radio similar to but different than SiriusXM

**Comment by gromgsxr** [comment]
audiobookshelf wold be awesome i have just discovered this and its perfect for me, i also found a python library for its api so i would assume that would help with implementing it maybe?

https://github.com/Teekeks/audiobookshelfAPI/tree/master

**Comment by AkoreJepade** [comment]
If possible I would like to see "Jango" added.

**Comment by scstraus** [comment]
I would like to see OwnTone (formerly ForkedDaapD) as a music provider. This allows me to import my iTunes/Apple music library and keep everything local and organized the way I want.

**Comment by jazzmonger** [comment]
from your latest release: "_Shortly after our last blog post, we added support for [Apple Music](https://www.music-assistant.io/music-providers/apple-music/) which was one of the most requested additions in the comments. A couple of other new providers were added, including [iBroadcast](https://www.music-assistant.io/music-providers/ibroadcast/) and [SiriusXM](https://www.music-assistant.io/music-providers/siriusxm/)_."

Now that the latest poll shows Pandora Support is the #1 request, I'm hopeful you'll listen to the community and we can finally get this implemented?  Its been a long time coming, and the ONLY thing preventing a good many of us from ditching our Alexa devices.  The "artist radio station" play mode is the best there is.   "Alexa Play Sting radio" or "Alexa Play Baroque radio" has some of the best curated selections of all the other providers.

**Reply by OzGav** [reply]
Pandora support will be added to 2.8.0 beta 5+ allowing playback of a users radio stations

**Reply by jazzmonger** [reply]
@OzGav I see Pandora was added in 2.8 beta 4, not 5?  Is this testable or just laying the foundation for 5+?
https://github.com/music-assistant/server/commit/5b5686281324eb71d5edf7dd56bf0c79a0ed290c

**Reply by OzGav** [reply]
No it is coming in 5 or you can use the nightly now.

**Comment by TGrounds** [comment]
I just implemented Music Assistant with Sonos and I found that I am keeping both the connection through Music Assistant but also my Sonos integration because I use Pandora through that integration.  I just voted and put Pandora as my number 1 followed by Amazon Music.  I'm loving the integrations and look forward to seeing how you address Pandora (along with Genera's).  Keep up the GREAT WORK!

**Reply by OzGav** [reply]
Pandora support will be added to 2.8.0 beta 5+ allowing playback of a users radio stations

**Comment by jem24h** [comment]
WebDAV is a universal tech ! We need it for cloud storage providers (such as pCloud, Nextcloud, ownCloud, Box, pCloud, Yandex.Disk, Koofr, Seafile, Mail.ru Cloud, 4shared, DriveHQ, MEGA)

**Reply by ntninja** [reply]
Maybe in next vote that list to the option listing, would probably get a lot more people interested in it.

**Comment by OBoudreaux** [comment]
Browser Source - There are some sources that I can only find on their respective webpages (e.g. the [americanroutes ](https://www.amroutes.org/)archive). It would be great to have something like browser_mod or fully kiosk as a music provider. This may already be possible but I can't seem to figure it out.

**Reply by OzGav** [reply]
No that would be tricky. Looking at the american routes archive you could contact them and find out whether it is ok to download their tracks as they are plainly visible on their webpages. You could then listen to them whenever you wanted...?

**Reply by OBoudreaux** [reply]
After digging into it further I see the difficulty.

How about using a pair ot ESP32's as an audio source? They are certainly cheap enough that using two isn't totally ridiculous (unless there is a variant that has separate BT/wifi radios).  One would act as a Bluetooth A2DP Sink, decode it and send it to the other that would broadcast the i2s stream over http.

I could see the utility of a generic esp32 source far beyond my particular use case.

**Reply by Hedda** [reply]
> I could see the utility of a generic esp32 source far beyond my particular use case.

There are indeed other similar use cases, see example these feature requests for analog AUX input -> https://github.com/orgs/music-assistant/discussions/2343 and https://github.com/music-assistant/tauri-companion-app/issues/40

**Reply by OBoudreaux** [reply]
I see that you can stream from a url in "tracks" or "Radio."

Might be able to pull this off with a single esp (though I'm not sure the time multiplexing would be all that great for audio. When I find some time I'll see if I can get this working.

**Comment by velodromen** [comment]
Is there any way to implement login details to radio stations which offers subscriptions for their premium radio stations and avoiding ads? One example I'm currently paying for is Klassik radio: https://select.klassikradio.de/en/what-is-select. 

I'm able to use MA to listen to the free stations offered by Klassik Radio. However the free stations are filled with ads (and the premium radio stations are a bit better), hence my question/request here.

**Reply by OzGav** [reply]
Someone would have to write a dedicated provider. We have SiriusXM for example.

**Reply by velodromen** [reply]
ok thanks, then I'll add Klassik radio to the wish list :)

**Comment by Hedda** [comment]
> It has been fourteen months since the last poll and a lot has changed. MA V2.0 is out of beta. Deezer, Jellyfin, Subsonic and Apple Music have been added as Music Providers. So it is time to re-do the poll to find out what the Community would like next!

@OzGav noticed that fourteen more months has now gone by since this last pull, and a bunch of additional music providers has been added, so wondering it is maybe time for another pull to guage community music provider feature request interests now? 😉

**Reply by Hedda** [reply]
My personal feature request for a new music provider would be to have an simple-to-use out-of-the-box Music Assistent AUX input appliance device solution based on ESPHome for streaming stereo audio that is easy for anyone to setup, see:

- https://github.com/orgs/music-assistant/discussions/2343

That feature request is close to the existing ” Local Audio In” concept but not quite the same. The goal of adding ability to digitizing analog audio and streaming it is the same though.

**Reply by OzGav** [reply]
Yes I'l re-do this one in the next few weeks. Local audio in has been started but time is the enemy,

**Comment by GrahamArden** [comment]
I'd love to see support for BBC Sounds (I realise that is a very UKcentric request). Currently it's the only feature missing that's stopping me from switxhing over completely from LMS.

**Reply by OzGav** [reply]
BBC Sounds is available in 2.7.0 beta 16 and later

**Comment by idar-a** [comment]
I would really like to see Storytel (www.storytel.no) as provider for audiobooks 🙌

**Comment by accma2** [comment]
UPnP servers, like un HA.

**Reply by Hedda** [reply]
FYI, there is an existing dedicated discussion about DLNA and UPnP sources here:

* https://github.com/orgs/music-assistant/discussions/890

**Comment by Hedda** [comment]
@OzGav can I suggest a new "Music Provider Feature Request Poll # 4" to start fresh again now that Music Assistant 2.7 has been released?

**Reply by Hedda** [reply]
@OzGav any thoughts on the suggestion of a new "Music Provider Feature Request Poll # 4" to start fresh again now that Music Assistant 2.7 has been released? …pin that as well as a new player provider poll too:

- https://github.com/orgs/music-assistant/discussions/3978

Would think that it should be interesting to get a head start in collecting that data before the next live stream and also ahead of the upcoming State of the Open Home from the Open Home Foundation now has a date in begining of April.

**Reply by OzGav** [reply]
Yeah I will do this one soon. The player provider poll I will leave for a while as there isn't that much to change.

**Comment by Hedda** [comment]
Btw, I would like to see a Sendspin Music Provider solution so Music Assistant server can also recieve an audio stream from a Sendspin source:

* https://github.com/orgs/music-assistant/discussions/4545
